src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -252,6 +256,32 @@ public sealed partial class TokenCache : ITokenCacheInternal
             }
         }
 
+        //This method pulls all of the acceess and refresh tokens from the cache and can therefore be very impactful on performance.
+        private void DumpCacheToLogs(AuthenticationRequestParameters requestParameters)
+        {
+            if (requestParameters.RequestContext.Logger.IsLoggingEnabled(LogLevel.Verbose))
+            {
+                IEnumerable<MsalAccessTokenCacheItem> accessTokenCacheItems = GetAllAccessTokensWithNoLocks(filterByClientId: false);
+                IReadOnlyList<MsalRefreshTokenCacheItem> refreshTokenCacheItems = GetAllRefreshTokensWithNoLocks(filterByClientId: false);
+                IEnumerable<MsalAccessTokenCacheKey> accessTokenCacheKeys = accessTokenCacheItems.Select(item => item.GetKey());
```

Comment:
Add a `.ToList()` and use the List instead of IEnumearble. To avoid enumerating multiple times.